yunwoo/baekjoon/큐/11866-요세푸스-문제-0.js

```diff
@@ -0,0 +1,39 @@
+const readline = require("readline");
+const rl = readline.createInterface({
+  input: process.stdin,
+  output: process.stdout,
+});
+
+const input = [];
+
+rl.on("line", function (line) {
+  input.push(line);
+}).on("close", function () {
+  // 풀이
+  const [n, k] = input[0].split(" ").map(Number);
+  const queue = Array.from({ length: n }, (_, i) => i + 1);
+  const result = [];
+  let count = 0;
+
+  while (queue.length !== 0) {
+    const people = queue.shift();
+
+    if (count !== k - 1) {
+      queue.push(people);
+    } else {
+      result.push(people);
+      count = 0;
+
+      continue;
+    }
+    count++;
+  }
+
+  console.log(`<${result.join(", ")}>`);
```

Comment:
```suggestion
  const [n, k] = input[0].split(" ").map(Number);
  const queue = Array.from({ length: n }, (_, i) => i + 1);
  const result = [];
  let count = 0;

  while (queue.length !== 0) {
    const removedFirstPeople = queue.shift();

    count++;

    if (count % k === 0) {
      result.push(removedFirstPeople);
    } else {
      queue.push(removedFirstPeople);
    }
  }

  console.log(`<${result.join(", ")}>`);
```
- 풀이 잘봤습니다 🙇‍♂️
- count를 증가시키는 위치와 조건식을 약간 변경해주면 continue와 같은 부분은 제거할 수 있습니다
- 다 좋은데 변수명에대해 좀 더 디테일하게 생각해보시는게 좋을 것 같습니다 (큐에서) 첫번째 사람을 제거하는 변수의 네이밍을 단순히 people이라고 하면 의미가 너무 명확하지가 않은 것 같아요 🥲
- 저는 우선 (큐에서) 제거되는 첫 번째 사람이라는 의미로 removedFirstPeople 라고 생각을 해봤는데, 윤우님께서도 지금보다 더~~!! 네이밍에 대해서 고민해보셨으면 좋겠습니다 !

Reply:
- 나머지 연산자를 이용하니까 count도 재할당 해줄 필요가없네요!!👍
- 변수명을 원래 선택된순서? 관련된거로 지으려고 찾아보다 못찾아서 요세푸스의 의미를 생각해서 사람으로 지었는데 어렵네요..🥲 좀 더 명확하게 지을 수 있도록 더 고민해보겠습니다!!!!!🔥